### PR TITLE
Initial draft of Pulsar rule addition

### PR DESCRIPTION
**Pull request by tbedford**
## Description

Adds Pulsar rule to Control API spec.

* [DOC-534](https://ably.atlassian.net/browse/DOC-534)

## Checklist for OpenAPI document updates

Please ensure the following:

- [x] Version has been incremented
- [x] Tested for errors with Spectral
- [ ] Optional: Load into SwaggerHub and check for errors
- [x] Ensure the document renders under ReDoc (see README for instructions on how to test locally)

## Review

Instructions on how to review the PR:

- [ ] Version has been incremented
- [ ] Tested for errors with Spectral
- [ ] Optional: Load into SwaggerHub and check for errors
- [ ] Ensure the document renders under ReDoc (see README for instructions on how to test locally)

### CODE REVIEW

File: definitions/control-v1.yaml [hunk 1]
```diff
@@ -3801,6 +3807,251 @@ components:
       - requestMode
       - source
       - target    
+    pulsar_token_auth:
+      type: object
+      description: Pulsar supports authenticating clients using security tokens that are based on JSON Web Tokens.
```
Comment (m-hulbert) [inline definitions/control-v1.yaml:3812]: This doesn't appear as the description for `authentication`. You can manually add it at as a description in situ e.g. line 3876

I know this is coming from the spec as well but it looks odd that Kafka uses `auth` and Pulsar uses `authentication`.

File: definitions/control-v1.yaml [hunk 2]
```diff
@@ -3801,6 +3807,251 @@ components:
...
+              nullable: true
+              example: 
+            authentication:
+              oneOf:
+              - "$ref": "#/components/schemas/pulsar_token_auth"
+              discriminator:
+                propertyName: authenticationMode
+                mapping:
+                  token: "#/components/schemas/pulsar_token_auth"
+            enveloped:
```
Comment (m-hulbert) [inline definitions/control-v1.yaml:3956]: Missing `description` and `example`.

File: definitions/control-v1.yaml [hunk 3]
```diff
@@ -3801,6 +3807,251 @@ components:
...
+              type: boolean
+              nullable: true
+            format:
```
Comment (m-hulbert) [inline definitions/control-v1.yaml:3959]: Missing `description` and `example`.

File: definitions/control-v1.yaml [hunk 4]
```diff
@@ -3801,6 +3807,251 @@ components:
...
+                type: string
+                example: "-----BEGIN CERTIFICATE-----
+MIIFiTCCA3GgAwIBAgIUYO1Lomxzj7VRawWwEFiQht9OLpUwDQYJKoZIhvcNAQEL
+BQAwTDELMAkGA1UEBhMCVVMxETAPBgNVBAgMCE1pY2hpZ2FuMQ8wDQYDVQQHDAZX
+...snip...
+TOfReTlUQzgpXRW5h3n2LVXbXQhPGcVitb88Cm2R8cxQwgB1VncM8yvmKhREo2tz
+7Y+sUx6eIl4dlNl9kVrH1TD3EwwtGsjUNlFSZhg=
+-----END CERTIFICATE-----"
+              nullable: true
+              example:
```
Comment (m-hulbert) [inline definitions/control-v1.yaml:3948]: This `example` isn't required and causes the `tlsTrustCerts` to be null in the example request.

**Review by m-hulbert — CHANGES_REQUESTED** [review]
Spotted a few missing/additional properties but looks good other than that.

File: definitions/control-v1.yaml (hunk 1 above)
Comment (tbedford) [reply definitions/control-v1.yaml:3812]: > This doesn't appear as the description for authentication. You can manually add it at as a description in situ e.g. line 3876

Well spotted. Fixed.

> I know this is coming from the spec as well but it looks odd that Kafka uses auth and Pulsar uses authentication.

Well spotted again. I'll raise a bug report for this. https://ably.atlassian.net/browse/WEB-2103

File: definitions/control-v1.yaml (hunk 4 above)
Comment (tbedford) [reply definitions/control-v1.yaml:3948]: Oops. Fixed.

File: definitions/control-v1.yaml (hunk 3 above)
Comment (tbedford) [reply definitions/control-v1.yaml:3959]: Well spotted - easily done :(

File: definitions/control-v1.yaml (hunk 2 above)
Comment (tbedford) [reply definitions/control-v1.yaml:3956]: Fixed.

File: definitions/control-v1.yaml (hunk 1 above)
Comment (m-hulbert) [reply definitions/control-v1.yaml]: Just need to do this one for PATCH as well I think :-)

Comment (tbedford) [reply definitions/control-v1.yaml]: ahh, quite right!

File: definitions/control-v1.yaml [hunk 5]
```diff
@@ -3801,6 +3807,256 @@ components:
       - requestMode
       - source
       - target    
+    pulsar_token_auth:
+      type: object
```
Comment (m-hulbert) [inline definitions/control-v1.yaml:3811]: Sorry... I just noticed that it's not picking up the type for `authentication` either(!)

Comment (tbedford) [reply definitions/control-v1.yaml:3811]: OK I will look into that! Thanks!

**Review by m-hulbert — APPROVED** [review]

**Pull request merged**